### ISSUE TITLE
Add new backends to seqrepo

### ISSUE THREAD

**Issue by korikuzma** [biocommons/hackathon#12]
### Submitter Name

Reece Hart (@reece)

### Submitter Affiliation

MyOme

### Requested By

seqrepo community

### Lead(s)

@reece 

### biocommons Repo

seqrepo

### Which day(s) would the project be offered?

- [ ] Hackathon Day 1
- [ ] Hackathon Day 2
- [ ] Hackathon Day 3

### Project Details

[Hackathon Project Slide](https://docs.google.com/presentation/d/13RZcV5NRpHwQkRfyhiezSh_bnayPntJr6IYkMDsdbzo/edit#slide=id.g25f4b8f0067_0_263)

[SeqRepo](https://github.com/biocommons/biocommons.seqrepo/) provides a non-redundant, compressed, journalled, file-based storage for biological sequences. SeqRepo also provides a simple interface to this storage. File-based storage is not convenient or efficient for many users. This project will generalize the interface to support other types of backends.

This project has several goals:
- Formalize an abstract interface to sequences and sequence slicing as described [here](https://github.com/biocommons/biocommons.seqrepo/issues/61) and implement it
- Investigate and implement additional backends, such as redis or s3, which might be preferable for public access
- Consider how to create a fingerprint of an instance of seqrepo data, such as digesting all sequences and aliases to create a fingerprint for the collection

**Comment by reece** [comment]
https://github.com/biocommons/biocommons.seqrepo/issues/61#issuecomment-1663176331 for the intended plan for generalizing the interface and implementing backends.

**Comment by korikuzma** [comment]
@ahwagner strongly agrees don't do the advance work and have this be part of the hackathon

**Comment by theferrit32** [comment]
@reece thanks for doing the investigation work here to confirm and providing the url and release numbers info. I loaded these files locally so I could dig around a bit. The downloads took a very long time.

I am imagining a user with an id like `ENST00000715724.1` would often just know it's Ensembl, so would want to resolve a generic namespaced id like `ensembl:ENST00000715724.1`, not knowing what specific release it was in. It would probably be useful to support this without needing custom client application code specific to Ensembl. Maybe there could be an option to tell SeqRepo to take the latest when there's conflicts? But that could be a problem for use cases like VOCA normalization if the changes in sequence between releases also changes the coordinates that a variant normalizes to.

**Comment by reece** [comment]
I agree that loading sequences into a single namespace seems pretty reasonable at this point. We can make that a goal for the hackathon.

**Comment by korikuzma** [comment]
@reece Would you be able to fill out the `Which day(s) would the project be offered?` to help determine how many day(s) this project will take?
